Vulkan: Keep state on Clear cmds when used later

unknownbrackets opened the pull request:
This also removes duplicate state - for example our UI has some duplicate Scissor commands.

Should solve the case from #10997 - and I found a game with a clear in the middle which I used to debug this.  But, still not sure about safety/risk for 1.6.0...

-[Unknown]

unknownbrackets committed:
Vulkan: Keep state on Clear cmds when used later.

This also removes duplicate state - for example our UI has some duplicate
Scissor commands.

unknownbrackets commented:
Made a tweak so it will remove duplicate state from before/after a clear too.  This ends up triggering sometimes (mostly transition screens) in various games, and even on every frame of Star Ocean on the world map.  Everything looks okay, though.

-[Unknown]

hrydgard commented:
Hm. I think it's cheaper to just switch the commands to REMOVED and ignore them during interpretation than to actually perform erase operations? For 1.6 I'd prefer a minimal version of this that just kills draw commands before a clear by marking them REMOVED and leaving everything else behind.

hrydgard commented on ext/native/thin3d/VulkanRenderManager.cpp:
```diff
@@ -612,6 +613,70 @@ bool VulkanRenderManager::InitDepthStencilBuffer(VkCommandBuffer cmd) {
+}
+
+static void CleanupRenderCommands(std::vector<VkRenderData> *cmds) {
+	size_t lastCommand[256];
```
Could just define a max value in the enum and use that here, 256 is too many for sure.

unknownbrackets replied:
I'm not sure it'll break the bank - wanted to avoid all the annoying default switch values for such enums.  Personally I think it's silly that c++11 still doesn't just have a max value automatically, especially for class enums.

-[Unknown]

hrydgard replied:
We do loop over this array at some point, it will eat a tiny bit of cpu :) but yeah memory-wise it shouldn't be a big deal... Meh.

unknownbrackets committed:
Vulkan: Leave removed commands in render passes.

We'll just need to not check commands.empty(), which we do in a couple
places but not in important ways so far.
